### ISSUE TITLE
Vertices method should accept generators

### ISSUE THREAD

**Issue by hx2A** [py5coding/py5generator#229]
![image](https://user-images.githubusercontent.com/3694604/216197226-d37cb3c2-3cc7-4c5a-9018-ec2d0aee911b.png)

```python
import trimesh
import shapely
# https://iamkate.com/data/12-bit-rainbow/
palette = (
    '#817', '#a35', '#c66', '#e94',
    '#ed0', '#9d5', '#4d8', '#2cb',
    '#0bc', '#09c', '#36b', '#639'
    )

def setup():
    global m
    size(400, 400, P3D)
    no_stroke()
    polygon = shapely.geometry.Polygon([(-100, -100), (0, -100),
                                        (0, 0), (-50, -50), (-100, 0)])
    m = trimesh.creation.extrude_polygon(polygon, 30)
      
def draw():
    background(0)
    translate(width /2, height / 2)
    rotate_x(QUARTER_PI)
    rotate_y(radians(mouse_x))
    for i, face in enumerate(m.faces):
        fill(palette[i % 12])
        with begin_closed_shape():
            vertices([m.vertices[v] for v in face])
  ```

@hx2A, would it be too hard to make `vertices()` accept a generator so we could avoid building that list?

_Originally posted by @villares in https://github.com/py5coding/py5generator/discussions/228_

**Comment by villares** [comment]
Do you think it would be a good idea to add that to `lines()`  and `points()` too?

**Comment by hx2A** [comment]
Yes, absolutely. Also the 3 curve methods (`bezier_vertices()`, etc).

**Comment by shyam-sunder-19** [comment]
So this requires a loop to iterate over the generator passed in as argument? I believe this needs to be done for vertices(), lines(), points() and the 3 curves. Lmk if I can proceed with this.

**Comment by hx2A** [comment]
> So this requires a loop to iterate over the generator passed in as argument? I believe this needs to be done for vertices(), lines(), points() and the 3 curves. Lmk if I can proceed with this.

No, no loops. You'll need to detect if the passed argument is a generator. If it is, convert it to a list, and then proceed with the existing code.

**Comment by hx2A** [comment]
I looked into this some more and I realized fixing this requires some extra complications that would be hard to explain. I removed the good first issue tag because it is not that.

**Comment by hx2A** [comment]
Fixed with #253
